iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2059,12 +2059,65 @@ TYPED_TEST(stringTyped_test, ConcatenateStringLiteralAndStringWorks)
 TEST(StringLiteralConcatenation, ConcatenateOnlyStringLiteralsWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "f591fb69-a8fa-4d8e-b42d-29573b4a20f7");
-    auto testString = iox::cxx::concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
+    auto testString = concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
     EXPECT_THAT(testString.capacity(), Eq(25U));
     EXPECT_THAT(testString.size(), Eq(25U));
     EXPECT_THAT(testString.c_str(), StrEq("FerdinandSpitzschnueffler"));
 }
 
+TYPED_TEST(stringTyped_test, ConcatenateEmptyStringAndNullCharReturnsStringWithSizeOne)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6a4374fb-b21f-4465-95dc-22b898079c81");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    auto result = concatenate(this->testSubject, '\0');
+
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+    EXPECT_THAT(this->testSubject.size(), Eq(0U));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
```

Comment:
I think this check is not necessary. We probably have a test already that checks that a size 0 string compares equal with `""`.

So checking that the size is still 0 should be sufficient (checking more does not hurt except for increased test-time, but is there a benefit?)